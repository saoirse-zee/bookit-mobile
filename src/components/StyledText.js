import React from 'react'
import { Text } from 'react-native'

export const MonoText = props => (
  <Text {...props} style={[props.style, { fontFamily: 'space-mono' }]} />
)

export const ShoutyText = props => (
  <Text
    {...props}
    style={[props.style, {
      color: '#555',
      fontSize: 32,
      marginBottom: 50,
    }]}
  />
)
