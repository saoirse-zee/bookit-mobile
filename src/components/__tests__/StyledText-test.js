import 'react-native'
import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'
import { MonoText, ShoutyText, LabelText } from '../StyledText'

it('renders MonoText correctly', () => {
  const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders ShoutyText correctly', () => {
  const tree = renderer.create(<ShoutyText>Snapshot test!</ShoutyText>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders LabelText correctly', () => {
  const tree = renderer.create(<LabelText>Snapshot test!</LabelText>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('respects optional styles for LabelText', () => {
  const tree = renderer.create(<LabelText style={{ color: 'tomato' }}>Snapshot test!</LabelText>).toJSON()
  expect(tree).toMatchSnapshot()
})
