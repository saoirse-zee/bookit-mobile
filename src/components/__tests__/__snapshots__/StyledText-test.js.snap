// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LabelText correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "color": "#555555",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`renders MonoText correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`renders ShoutyText correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "color": "#555",
        "fontSize": 32,
        "marginBottom": 50,
      },
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`respects optional styles for LabelText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "tomato",
      },
      Object {
        "color": "#555555",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;
